contracts/InterchainTokenFactory.sol

```diff
@@ -152,19 +172,73 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
         }
     }
 
+    /**
+     * @notice Allow the minter to approve the deployer for a remote interchain token deployment that uses a custom destinationMinter address.
+     * This ensures that a token deployer can't choose the destinationMinter itself, and requires the approval of the minter to reduce trust assumptions on the deployer.
+     */
+    function approveDeployRemoteInterchainToken(address deployer, bytes32 salt, string calldata destinationChain, bytes calldata destinationMinter) external {
```

Comment:
Anyone can call the `approveDeployRemoteInterchainToken` function. Would it be safe to derive the `tokenId` from `salt` and check `if (!token.isMinter(msg.sender)) revert NotMinter(msg.sender);` to make the approval `true? This might be overkill since the `if (!token.isMinter(minter)) revert NotMinter(minter);` check is already present in the `deployRemoteInterchainToken` function, but I think it’s good to ensure there are no security concerns.

Reply:
fair point, we can add the check, and since the gas cost isn't a concern. What we should ensure is to not remove the same check in deploy* because the minter could be transferred later, and the approval is stale. i'll add a comment for it